### DISCUSSION TITLE
Is it possible to share a single raw TCP socket(or other transport) between different SPs?

### DISCUSSION THREAD

**Discussion by helloworld00** [nanomsg/nng#1876] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I want to build a server app that listen on a port. And it can accept SPs or HTTP connections from different clients on that port. 

**Describe the solution you'd like**
AFAIK, the "transport" layer is responsible for transmit bytes array. The SPs are "application protocol" layer, like HTTP. When the server accepts the client transport connection, is it possible to identify the "application protocol" from the initial bytes? So the "application protocol" is specific to one client connection.

**Describe alternatives you've considered**
None

**Additional context**
None

**Comment by gdamore** [comment, marked as answer]
Generally, "no".

But if you want to share a *port* (e.g. TCP port XYZ), then you can do so if you use websocket.  Right now that's the only way to do it.

**Comment by gdamore** [comment]
In fact, the websocket transported is designed for HTTP and SP coexistence!

**Comment by helloworld00** [comment]
Thanks!
